sdc/datatypes/hpat_pandas_groupby_functions.py

```diff
@@ -233,6 +235,174 @@ def _sdc_pandas_groupby_generic_func_codegen(func_name, columns, func_params, de
+        f'  accums_parts_{i} = [List.empty_list(accums_dtype_{i}) for _ in range(chunks_num)]',
+    ]) for i, *_ in column_dtype_indices.values()]
+    func_lines += [
+        f'  for i in prange(chunks_num):',
+        f'    chunk = chunks[i]',
+        f'    idx = 0',
+        f'    index_labels = index_labels_parts[i]',
+        f'    label_indices = label_indices_parts[i]',
+    ]
+    func_lines += ['\n'.join([
+        f'    label_accums_{i} = label_accums_parts_{i}[i]',
+        f'    accums_{i} = accums_parts_{i}[i]',
+        f'    accums_{i}.append(numpy.zeros({len(tail)+1}, dtype={ty}))'
+    ]) for ty, (i, *tail) in column_dtype_indices.items()]
+    func_lines += [
+        f'    for k in range(chunk.start, chunk.stop):',
+        f'      label = by_column_data[k]',
+    ]
+    func_lines += ['\n'.join([
+        f'      _accums_{i} = label_accums_{i}.get(label)',
+    ]) for i, *_ in column_dtype_indices.values()]
+    func_lines += [
+        f'      if _accums_0 is None:',
+    ]
+    func_lines += ['\n'.join([
+        f'        new_accums_{i} = numpy.zeros({len(tail)+1}, dtype={ty})',
+        f'        _accums_{i} = new_accums_{i}',
+        f'        label_accums_{i}[label] = new_accums_{i}',
+    ]) for ty, (i, *tail) in column_dtype_indices.items()]
+    func_lines += [
+        f'        index_labels[idx] = label',
+        f'        label_indices[label] = idx',
+        f'        idx += 1',
+    ]
+    func_lines += ['\n'.join([
+        f'      val_{idx} = column_data_{idx}[k]',
+        f'      if not numpy.isnan(val_{idx}):',
+        f'        _accums_{indices[0]}[{i}] += val_{idx}',
+    ]) for indices in column_dtype_indices.values() for i, idx in enumerate(indices)]
+    func_lines += ['\n'.join([
+        f'      accums_{i}[0] = _accums_{i}',
+    ]) for i, *_ in column_dtype_indices.values()]
+    func_lines += [
+        f'  def reduce_indices(main_labels, auxiliary_labels, main_indices):',
+        f'    idx = len(main_indices)',
+        f'    for _idx in range(len(auxiliary_labels)):',
+        f'      label = auxiliary_labels[_idx]',
+        f'      if label in main_indices:',
+        f'        continue',
+        f'      main_labels[idx] = label',
+        f'      main_indices[label] = idx',
+        f'      idx += 1',
+        f'  def reduce_accums(main_accums, auxiliary_accums, accums):',
+        f'    for label, _acc in auxiliary_accums.items():',
+        f'      _accums = main_accums.get(label)',
+        f'      if _accums is None:',
+        f'        _accums = _acc',
+        f'        main_accums[label] = _accums',
+        f'      else:',
+        f'        for i in range(len(_accums)):',
+        f'          _accums[i] += _acc[i]',
+        f'      accums[0] = _accums',
+        f'  shift = 1',
+        f'  while shift < chunks_num:',
+        f'    step = shift * 2',
+        f'    stop = int(math.ceil((chunks_num - shift) / step))',
```

Comment:
You can probably use np.ceil (since numpy is imported anyway) and not used math module at all.

Reply:
Thanks, replaced `math.ceil` with `np.ceil`.